docs/secret_management.md

```diff
@@ -0,0 +1,172 @@
+# Secrets Management for the API and APP keys
+
+The Datadog-Operator is able to retrieves the datadog credentials securely thanks to 3 different methods.
+
+## 1. Plain credentials in DatadogAgent resource
+
+this is the simplest way to provide the credentials that will be used by the agents.
+
+```yaml
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogAgent
+metadata:
+  name: datadog
+spec:
+  credentials:
+    apiKey: "<api-key>"
+    appKey: "<app-key>"
+  # ...
+```
+
+The credentials provided here will be stored in a Secret created by the Operator. By setting properly the `RBAC` on the `DatadogAgent` CRD. It is possible to limit who is able to see those credentials.
+But still, it is not the best solution in terms of security. This solution is good for testing purposes.
+
+## 2. Use secret(s) references
+
+Another solution is to provide the name of the secret(s) that store the credentials. like this the `DatadogAgent` resource doesn't contain any credentials as plain text.
+
+```yaml
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogAgent
+metadata:
+  name: datadog
+spec:
+  credentials:
+    apiKeyExistingSecret: "my-api-key-secret"
+    appKeyExistingSecret: "my-app-key-secret"
+  # ...
+```
+
+In this case, the secret(s) should exist before trying to create the `DatadogAgent`, else the deployment will failed.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-api-key-secret
+data:
+  api_key: <api-key>
+
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-app-key-secret
+data:
+  app_key: <app-key>
+```
+
+The keys used in the secret(s) are important. for the "API key" the key inside the secret should be `api_key`, and for the "APP key" it is `app_key`.
```

Comment:
When I read this I thought we were referring to the credentials not the data keys in the secret
```suggestion
The data keys used in the secret(s) are important. for the "API key" the key inside the secret should be `api_key`, and for the "APP key" it is `app_key`.
```